src/task/file_providers/http_file_provider.rs

```diff
@@ -1,17 +1,42 @@
 use std::path::PathBuf;
 
-use crate::{file, http::HTTP};
+use md5::Digest;
+use sha2::Sha256;
+
+use crate::{env, file, http::HTTP};
 
 use super::TaskFileProvider;
 
 #[derive(Debug)]
 pub struct HttpTaskFileProvider {
-    tmpdir: PathBuf,
+    cache_path: PathBuf,
+    no_cache: bool,
 }
 
 impl HttpTaskFileProvider {
-    pub fn new(tmpdir: PathBuf) -> Self {
-        Self { tmpdir }
+    pub fn new(cache_path: PathBuf, no_cache: bool) -> Self {
+        Self {
+            cache_path,
+            no_cache,
+        }
+    }
+}
+
+impl HttpTaskFileProvider {
```

Comment:
I'd like to rename this and similar names from "FileProvider" to "RemoteTask" I think

```suggestion
impl RemoteTaskHttp {
```